Lib/cProfile.py

```diff
@@ -170,7 +171,12 @@ def main():
                 '__package__': None,
                 '__cached__': None,
             }
-        runctx(code, globs, None, options.outfile, options.sort)
+        try:
+            runctx(code, globs, None, options.outfile, options.sort)
+        except BrokenPipeError as exc:
+            # Prevent "Exception ignored" during interpreter shutdown.
+            sys.stdout = sys.stderr = io.StringIO()
```

Comment:
Setting stderr to a new stringIO can hide other errors written into stderr at exit. 

BrokenPipeError is more likely related to stdout than stderr. I understand that your intent is to ignore the warning displayed on stderr when sys.stdout file is closed.

There is now sys.unraisablehook which can be overidden temporarily to not log the exception on stderr when stdout is closed.

Since json.tool already has a similar code pattern, maybe it would be worth it to provide a helper function somewhere. I would prefer a private function.

By the way, it might be better to delete sys.stdout or set it to None, rather than creating a StringIO. So if the code after that tries to write tons of lines into sys.stdout, the writes are just ignored/lost, rather than growing memory (which will never be consumed anyway).